plugins/modules/docker_config.py

```diff
@@ -316,17 +331,20 @@ def present(self):
             self.results['config_id'] = config['ID']
             self.results['config_name'] = config['Spec']['Name']
             data_changed = False
+            template_driver_changed = False
             attrs = config.get('Spec', {})
             if attrs.get('Labels', {}).get('ansible_key'):
                 if attrs['Labels']['ansible_key'] != self.data_key:
                     data_changed = True
             else:
                 if not self.force:
                     self.client.module.warn("'ansible_key' label not found. Config will not be changed unless the force parameter is set to 'yes'")
+            if attrs['Labels']['template_driver'] != self.template_driver:
```

Comment:
This should likely be something like:
```suggestion
            if attrs.get(...) != self.template_driver:
```
Unfortunately the API docs (https://docs.docker.com/engine/api/v1.41/#operation/ConfigInspect) do not give any hint how and where that information is available. Can you paste the output of something like `docker config inspect ...` for such a config?

In any case, this is one more reason why there should be tests for this in CI :)

Reply:
For this docker config task
```
- name: debug
  community.docker.docker_config:
    name: db_password
    data: opensesame!
    state: present
    template_driver: golang
```
this output of `docker config inspect ...` looks like this
```
[root@447f901529a4 /]# docker config inspect db_password
[
    {
        "ID": "ilwv23ojrt2ooisz06enkmovj",
        "Version": {
            "Index": 28681
        },
        "CreatedAt": "2022-05-11T01:45:58.5768652Z",
        "UpdatedAt": "2022-05-11T01:45:58.5768652Z",
        "Spec": {
            "Name": "db_password",
            "Labels": {
                "ansible_key": "4c8ed891d2bcad5bd4359a985e82349ec42e0c04be5511b0e53ac258"
            },
            "Data": "b3BlbnNlc2FtZSE=",
            "Templating": {
                "Name": "golang"
            }
        }
    }
]
```
Hence, I now access the `attrs['Templating']['Name']` to get a config's `template_driver`. I have done some local testing and it seems to work. I will now work on adding integration tests.